src/Position.js

```diff
@@ -0,0 +1,77 @@
+import React, { cloneElement } from 'react';
+import domUtils from './utils/domUtils';
+import { calcOverlayPosition } from './utils/overlayPositionUtils';
+import CustomPropTypes from './utils/CustomPropTypes';
+
+class PositionedComponent extends React.Component {
+
+  constructor(props, context){
+    super(props, context);
+    this.state = {
+      positionLeft: null,
+      positionTop: null,
+      arrowOffsetLeft: null,
+      arrowOffsetTop: null
+    };
+  }
+
+  componentWillMount(){
+    this._needsflush = true;
+  }
+
+  componentWillRecieveProps(){
+    this._needsflush = true;
+  }
+
+  componentDidMount(){
+    if ( !this._needsflush ) { return; }
```

Comment:
Maybe just DRY this and move the check into `_updatePosition`? Or call it `_maybeUpdatePosition` or whatever.